### DISCUSSION TITLE
How to completely remove and re-add the custom integration?

### DISCUSSION THREAD

**Discussion by N3rd7** [jbergler/hass-ttlock#144] [Q&A]
I'm experiencing difficulties signing in and have been tracking a similar open issue. However, my situation seems slightly different. Initially, when I added the TTlock integration, I may have mistakenly entered incorrect OAuth details on the initial screen. Now, although TTlock appears as a custom repository, I haven't completed the setup due to login issues on the subsequent screen, which requires app credentials. 
Suspecting that the wrong info entered on the first screen is the culprit, I attempted to uninstall and reinstall the HACS repository, restarting Home Assistant (HA) in the process. Yet, each time I try to add the integration, it bypasses the initial screen and takes me directly to the second screen for the TTlock login. 
What steps should I take to perform a "clean install" of the HACS extension, or somehow clear the cache, to rectify the OAuth information?

**Comment by jbergler** [comment]
Hiya, the oauth creds are stored using home assistants builtin "application credentials" feature, which is hidden in a menu from the devices page.
Docs are here: https://www.home-assistant.io/integrations/application_credentials/

Hope that helps.

**Comment by N3rd7** [comment]
This fully solved my issue, thank you! Updated with the correct Client ID and secret, and the next screen let me log in with the app credentials. Thank you very much!
